homeassistant/components/backup/backup.py

```diff
@@ -0,0 +1,146 @@
+        """Read backups from disk."""
+        backups: dict[str, LocalBackup] = {}
+        for backup_path in self.backup_dir.glob("*.tar"):
+            try:
+                with tarfile.open(backup_path, "r:", bufsize=BUF_SIZE) as backup_file:
+                    if data_file := backup_file.extractfile("./backup.json"):
+                        data = json_loads_object(data_file.read())
+                        backup = LocalBackup(
+                            id=cast(str, data["slug"]),  # Do we need another ID?
+                            slug=cast(str, data["slug"]),
+                            name=cast(str, data["name"]),
+                            date=cast(str, data["date"]),
+                            path=backup_path,
+                            size=round(backup_path.stat().st_size / 1_048_576, 2),
+                            protected=cast(bool, data.get("protected", False)),
+                        )
+                        backups[backup.slug] = backup
+            except (OSError, TarError, json.JSONDecodeError, KeyError) as err:
+                LOGGER.warning("Unable to read backup %s: %s", backup_path, err)
+        return backups
+
+    async def async_download_backup(
+        self,
+        *,
+        id: str,
+        path: Path,
+        **kwargs: Any,
+    ) -> None:
+        """Download a backup file."""
+        raise NotImplementedError
+
+    async def async_upload_backup(
+        self,
+        *,
+        path: Path,
+        metadata: BackupUploadMetadata,
+        **kwargs: Any,
+    ) -> None:
+        """Upload a backup."""
+        self.backups[metadata.slug] = LocalBackup(
+            id=metadata.slug,  # Do we need another ID?
+            slug=metadata.slug,
+            name=metadata.name,
+            date=metadata.date,
+            path=path,
+            size=round(path.stat().st_size / 1_048_576, 2),
+            protected=metadata.protected,
+        )
+
+    async def async_list_backups(self, **kwargs: Any) -> list[UploadedBackup]:
+        """List backups."""
+        if not self.loaded_backups:
+            await self.load_backups()
+        return list(self.backups.values())
+
+    async def async_get_backup(
+        self, *, slug: str, **kwargs: Any
+    ) -> UploadedBackup | None:
+        """Return a backup."""
+        if not self.loaded_backups:
+            await self.load_backups()
+
+        if not (backup := self.backups.get(slug)):
+            return None
+
+        if not backup.path.exists():
+            LOGGER.debug(
+                (
+                    "Removing tracked backup (%s) that does not exists on the expected"
+                    " path %s"
+                ),
+                backup.slug,
+                backup.path,
+            )
+            self.backups.pop(slug)
+            return None
+
+        return backup
+
+    async def async_remove_backup(self, *, slug: str, **kwargs: Any) -> None:
```

Comment:
This isn't part of the agent interface yet. I'm working on adding `async_delete_backup`.

Reply:
We can keep it for now. I'll replace it later.